src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Services/SyntaxFacts/CSharpHeaderFacts.cs

```diff
@@ -28,11 +28,33 @@ public override bool IsOnTypeHeader(SyntaxNode root, int position, bool fullHead
         if (node == null)
             return false;
 
-        var lastToken = (node as TypeDeclarationSyntax)?.TypeParameterList?.GetLastToken() ?? node.Identifier;
-        if (fullHeader)
-            lastToken = node.BaseList?.GetLastToken() ?? lastToken;
+        return IsOnHeader(root, position, node, GetLastToken());
 
-        return IsOnHeader(root, position, node, lastToken);
+        SyntaxToken GetLastToken()
+        {
+            if (fullHeader && node.BaseList != null)
+                return node.BaseList.GetLastToken();
+
+            if (node is TypeDeclarationSyntax typeDeclaration)
+            {
+                if (typeDeclaration.TypeParameterList != null)
+                    return typeDeclaration.TypeParameterList.GreaterThanToken;
+
+                // .Identifier may be default in the case of an extension type.
+                if (typeDeclaration.Identifier != default)
+                    return typeDeclaration.Identifier;
+
+                return typeDeclaration.Keyword;
+            }
+            else if (node is EnumDeclarationSyntax enumDeclaration)
```

Comment:
>else if (node is EnumDeclarationSyntax enumDeclaration) [](http://example.com/codeflow?start=12&length=55)

I'm not following this, is this a new case lighting up for these?

Reply:
The `EnumDeclarationSyntax` case would have been covered by the `?? node.Identifier;` in the original code. The change made it more readable to me.